### ISSUE TITLE
Exception support in prange body

### ISSUE THREAD

**Issue by ehsantn** [numba/numba#2811]
prange doesn't support exceptions in body since the loop can have multiple exits (check done [here](https://github.com/numba/numba/blob/master/numba/parfor.py#L810)). However, at least the exception case can be handled I think. Is there a problem with threads raising exceptions?

```python
import numpy as np
from numba import njit, prange

@njit(parallel=True)
def f(X, N):
    acc = 0
    for i in prange(N):
        assert X[i] == 1
        acc += X[i]
    return acc

N = 9
X = np.ones(N)
print(f(X, N))
```

**Comment by sklam** [comment]
Yes, parallel ufunc is not propagating an error.  There is a stalled patch: https://github.com/numba/numba/pull/2307

**Comment by slayoo** [comment]
Would be great if a warning would be issued (during compilation) if parallel=True is used and an exception is being raised from within a parallel block
